query-core/src/main/java/org/infinispan/query/core/impl/AggregatingQuery.java

```diff
@@ -47,9 +47,18 @@ public AggregatingQuery(QueryFactory queryFactory, AdvancedCache<?, ?> cache, St
    @Override
    protected Iterator<?> getBaseIterator() {
       Grouper grouper = new Grouper(noOfGroupingColumns, accumulators, twoPhaseAcc);
-      List<Object[]> list = baseQuery.list();
-      for (Object[] row : list) {
-         grouper.addRow(row);
+      List<Object> list = baseQuery.list();
+      for (Object row : list) {
+         // TODO HSEARCH-1807 maybe it is better to use a rowProcessor in the EmbeddedLuceneQuery instead ?
```

Comment:
As you probably have guessed, this depends on the fact that Hibernate Search 6 does not wrap single item projections and returns a list for multi item projections, while Infinispan here always expects a list of arrays (for both cases). Aggregation is built on projection results directly by Infinispan, without using Hibernate Search aggregation features. I agree that doing this conversion here is quite bad, since we haven't touched Infinispan core module so far. I’m trying to move the conversion to a previous point.

Reply:
Done!